Analysis/Tasks/vertexerhf.cxx

```diff
@@ -48,8 +48,10 @@ struct DecayVertexBuilder2Prong {
     hvtxp_x_out->Fill(collision.posX());
     hvtxp_y_out->Fill(collision.posY());
     hvtxp_z_out->Fill(collision.posZ());
-    for (auto it_0 = tracks.begin(); it_0 != tracks.end(); ++it_0) {
-      auto& track_0 = *it_0;
+    auto comb = o2::soa::CombinationsGenerator<soa::Join<aod::Tracks, aod::TracksCov, aod::TracksExtra>, 2>(tracks,  { return true; });
+    for (auto trackPair : comb) {
+      auto& track_0 = trackPair[0];
+      auto& track_1 = trackPair[1];
```

Comment:
@saganatt @ginnocen shouldn't this become:

```C++
for (auto [track_0, track_1] : generators(tracks, [](){ return true; }))
```

Reply:
We have this nice wrapper with combinations() in my current PR, so it will be prettier soon, once my PR gets merged.

Reply:
hi @ktf @jgrosseo I confirm that this gives the same results as the traditional double loop. I let you decide if you want to merge this one already or wait for @saganatt's PR to merge already the new version